### PR TITLE
Fix/compose filter logic

### PR DESCRIPTION
**Pull request by filbabic**
### 🎯 Goal

Applying the same logic for Compose filters as with XML.

#### Default Behavior
1. Run the app with default ChannelScreen.
2. You should see messaging channels where the current user is a member.

#### Custom initial filters
1. Pass in a custom filter to the channels screen.
2. You should see the data according to the custom filters.
4. There should be no data updates (e.g. filter changes) with time.

<details>
  <summary>Patch file to apply these changes for testing</summary>

```
Index: stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/ChannelsActivity.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/ChannelsActivity.kt b/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/ChannelsActivity.kt
--- a/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/ChannelsActivity.kt	(revision c441694aa06c69411ce9e559f9ce2ba52270d21e)
+++ b/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/ChannelsActivity.kt	(date 1651151940185)
@@ -47,6 +47,7 @@
 import io.getstream.chat.android.client.ChatClient
 import io.getstream.chat.android.client.api.models.QuerySort
 import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.Filters
 import io.getstream.chat.android.client.models.User
 import io.getstream.chat.android.compose.sample.ChatApp
 import io.getstream.chat.android.compose.sample.ChatHelper
@@ -89,6 +90,10 @@
         setContent {
             ChatTheme(dateFormatter = ChatApp.dateFormatter) {
                 ChannelsScreen(
+                    filters = Filters.and(
+                        Filters.eq("type", "messaging"),
+                        Filters.nin("members", listOf("dmitrii"))
+                    ),
                     title = stringResource(id = R.string.app_name),
                     isShowingHeader = true,
                     isShowingSearch = true,

```

</details>

#### Changing filters in the runtime (from default)
1. Run the app with the default filters.
2. Change filters after a while.
3. You should see data change with the filter. No data changes should happen in the future (based on filter).

<details>
  <summary>Patch file to apply these changes for testing</summary>

```
Index: stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/ChannelsActivity.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/ChannelsActivity.kt b/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/ChannelsActivity.kt
--- a/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/ChannelsActivity.kt	(revision c441694aa06c69411ce9e559f9ce2ba52270d21e)
+++ b/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/ChannelsActivity.kt	(date 1651151940185)
@@ -35,10 +35,12 @@
 import androidx.compose.material.Scaffold
 import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.collectAsState
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
@@ -47,9 +49,9 @@
 import io.getstream.chat.android.client.ChatClient
 import io.getstream.chat.android.client.api.models.QuerySort
 import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.Filters
 import io.getstream.chat.android.client.models.User
 import io.getstream.chat.android.compose.sample.ChatApp
-import io.getstream.chat.android.compose.sample.ChatHelper
 import io.getstream.chat.android.compose.sample.R
 import io.getstream.chat.android.compose.sample.ui.login.UserLoginActivity
 import io.getstream.chat.android.compose.state.channels.list.ChannelItemState
@@ -63,6 +65,8 @@
 import io.getstream.chat.android.compose.viewmodel.channels.ChannelListViewModel
 import io.getstream.chat.android.compose.viewmodel.channels.ChannelViewModelFactory
 import io.getstream.chat.android.offline.extensions.globalState
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
 
 class ChannelsActivity : BaseConnectedActivity() {
 
@@ -88,21 +92,21 @@
          */
         setContent {
             ChatTheme(dateFormatter = ChatApp.dateFormatter) {
-                ChannelsScreen(
-                    title = stringResource(id = R.string.app_name),
-                    isShowingHeader = true,
-                    isShowingSearch = true,
-                    onItemClick = ::openMessages,
-                    onBackPressed = ::finish,
-                    onHeaderAvatarClick = {
-                        ChatHelper.disconnectUser()
-
-                        openUserLogin()
-                    }
-                )
+                // ChannelsScreen(
+                //     title = stringResource(id = R.string.app_name),
+                //     isShowingHeader = true,
+                //     isShowingSearch = true,
+                //     onItemClick = ::openMessages,
+                //     onBackPressed = ::finish,
+                //     onHeaderAvatarClick = {
+                //         ChatHelper.disconnectUser()
+                //
+                //         openUserLogin()
+                //     }
+                // )
 
 //                MyCustomUiSimplified()
-//                MyCustomUi()
+                MyCustomUi()
             }
         }
     }
@@ -225,6 +229,20 @@
                 )
             }
         }
+
+        val scope = rememberCoroutineScope()
+
+        LaunchedEffect(Unit) {
+            scope.launch {
+                delay(2000)
+                listViewModel.setFilters(
+                    Filters.and(
+                        Filters.eq("type", "messaging"),
+                        Filters.nin("members", listOf("filip"))
+                    )
+                )
+            }
+        }
     }
 
     private fun openMessages(channel: Channel) {

```

</details>

#### General
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channels/ChannelListViewModel.kt [hunk 1]
```diff
@@ -146,22 +176,19 @@ public class ChannelListViewModel(
...
-            val queryChannelsRequest = QueryChannelsRequest(
-                filter = filter,
-                querySort = config.querySort,
-                limit = channelLimit,
-                messageLimit = messageLimit,
-                memberLimit = memberLimit,
-            )
+        searchQuery.combine(queryConfigFlow) { query, config -> query to config }
+            .collectLatest { (query, config) ->
+                val queryChannelsRequest = QueryChannelsRequest(
```
Comment (bychkovdmitry) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channels/ChannelListViewModel.kt:181]: Sometime we get 2 emits here:
- The first one with NeutralFilterObject
- The second one with the correct default filter
Can we don something about that?

Comment (filbabic) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channels/ChannelListViewModel.kt:181]: Good point - we could filterNotNull on the `filter` flow and then just ignore any "requests" where the filter doesn't exist

Comment (filbabic) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channels/ChannelListViewModel.kt:180]: @bychkovdmitry added a filter! Can you check now? :]

**Review by bychkovdmitry — APPROVED** [review]
LGTM

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channels/ChannelListViewModel.kt (hunk 1 above)
Comment (bychkovdmitry) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channels/ChannelListViewModel.kt:180]: The issue is solved

**Review by adasiewiczr — APPROVED** [review]
### ☑️Reviewer Checklist
- [x] Compose sample runs & works
- [x] Bugs validated (bugfixes)
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Pull request merged**